Go version missing from documentations.

I'm using golang v1.12 but this package is not supported because of errors.Is and errors.As functions which are not present in v1.12 and was added in v1.13. And there is no version requirement mentioned in the documentation. for this package

Having random errors like

```
> go get -u github.com/mattermost/mmctl
# github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror
go/src/github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror/multierror.go:112:9: undefined: errors.As
go/src/github.com/mattermost/mmctl/vendor/github.com/hashicorp/go-multierror/multierror.go:117:9: undefined: errors.Is
```

How does one know you need to upgrade go? I have no idea how package mangagement works in Go, but I'm used to version constraint explanations in other langs/package managers (js/php) etc.

Hmm, yeah this should be documented. FWIW, v1.0.0 supports go < 1.13, while  v1.1.0 requires go 1.13.

I also notice that in the go.mod file, the line `go 1.14`. This indicates that the module requires go 1.14. Infact, when I build this using go 1.12, I get the following error:

```
root@f97226dc12fd:/go/src/github.com/mwhooker/testmultierror# go run main.go
go: finding github.com/hashicorp/go-multierror v1.1.0
go: downloading github.com/hashicorp/go-multierror v1.1.0
go: extracting github.com/hashicorp/go-multierror v1.1.0
go: finding github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
# github.com/hashicorp/go-multierror
/go/pkg/mod/github.com/hashicorp/go-multierror@v1.1.0/multierror.go:112:9: undefined: errors.As
/go/pkg/mod/github.com/hashicorp/go-multierror@v1.1.0/multierror.go:117:9: undefined: errors.Is
**note: module requires Go 1.14**
```

(emphasis mine). 

Of course, because there's this ticket, this doesn't seem sufficient, but the  requirement does exist. I would be happy to review any PRs that came in for documenting the  required go version

--- a/README.md
+++ b/README.md
@@ -26,6 +26,24 @@ Install using `go get github.com/hashicorp/go-multierror`.
 Full documentation is available at
 http://godoc.org/github.com/hashicorp/go-multierror
 
+### Requires go version 1.13 or newer
+
+`go-multierror` requires go version 1.13 or newer. Go 1.13 introduced
+[error wrapping](https://golang.org/doc/go1.13#error_wrapping), which
+this library takes advantage of.
+
+If you need to use an earlier version of go, you can use the
+[v1.0.0](https://github.com/hashicorp/go-multierror/tree/v1.0.0)
+tag, which doesn't rely on features in go 1.13.
+
+If you see compile errors that look like the below, it's likely that
+you're on an older version of go.
+
+```
+/go/src/github.com/hashicorp/go-multierror/multierror.go:112:9: undefined: errors.As
+/go/src/github.com/hashicorp/go-multierror/multierror.go:117:9: undefined: errors.Is
+```
+
 ## Usage
 
 go-multierror is easy to use and purposely built to be unobtrusive in

--- a/README.md
+++ b/README.md
@@ -37,7 +37,7 @@ If you need to use an earlier version of go, you can use the
 tag, which doesn't rely on features in go 1.13.
 
 If you see compile errors that look like the below, it's likely that
-you're on an older version of go.
+you're on an older version of go:
 
 ```
 /go/src/github.com/hashicorp/go-multierror/multierror.go:112:9: undefined: errors.As

